### ISSUE TITLE
using oradb::autostartdatabase for multiple databases per server failing

### ISSUE THREAD

**Issue by sureshr7** [biemond/biemond-oradb#267]
Hi,
  
oradb::autostartdatabase { 'DEV':
      oracle_home  => '/u01/oracle/product/12.1.0.2/db',
      db_name      => 'DEV',
      user         => 'oracle',
      service_name => 'DEV',
    }
    oradb::autostartdatabase { 'STG':
      oracle_home  => '/u01/oracle/product/12.1.0.2/dbstage',
      db_name      => 'STG',
      user         => 'oracle',
      service_name => 'STG',
    }
Using above code on a single server fails with error message 

Error while evaluating a Resource Statement, Duplicate declaration: Class[Oradb::Prepareautostart] is already declared at (file: modules/oradb/manifests/autostartdatabase.pp, line: 27); cannot redeclare (file:modules/oradb/manifests/autostartdatabase.pp, line: 27)

Thanks
Suresh

**Comment by sureshr7** [comment]
Looks like using virtual resource solves this problem. I will test and confirm

@oradb::autostartdatabase { 'DEV':
oracle_home => '/u01/oracle/product/12.1.0.2/db',
db_name => 'DEV',
user => 'oracle',
service_name => 'DEV',
}
@oradb::autostartdatabase { 'STG':
oracle_home => '/u01/oracle/product/12.1.0.2/dbstage',
db_name => 'STG',
user => 'oracle',
service_name => 'STG',
}

**Comment by sureshr7** [comment]
Still its failing with same error if i use realize in the class

**Comment by biemond** [comment]
yep, need to make a tempate Prepareautostart  for this, class can only exists once.

**Comment by sureshr7** [comment]
Thanks for the quick ack Edwin. Any workarounds to handle this ?

**Comment by biemond** [comment]
I think I need to make this a define https://github.com/biemond/biemond-oradb/blob/puppet4_3_data/manifests/prepareautostart.pp 
but maybe you only need to define it once and update oratab to y for the 2nd instance.  I think the service will start both.

**Comment by sureshr7** [comment]
yes. You are correct.
 making https://github.com/biemond/biemond-oradb/blob/puppet4_3_data/manifests/prepareautostart.pp as define and removing the class call in https://github.com/biemond/biemond-oradb/blob/puppet4_3_data/manifests/autostartdatabase.pp fixes the problem

Call both resources explicity  
   oradb::autostartdatabase { $db_name:
      oracle_home  => $oracle_home,
      db_name      => $db_name,
      user         => $user,
      service_name => $db_name,
   }

   oradb::prepareautostart{ $db_name:
    oracle_home  => $oracle_home,
    user         => $user,
    service_name => $db_name,
  }

**Comment by sureshr7** [comment]
Hi Edwin,

Is it possible to push this fix ? 

Thanks
Suresh
